src/acore/enrichment_analysis.py

```diff
@@ -353,11 +360,17 @@ def run_regulation_enrichment(
          )
     """
     # ? can we remove NA features in that column?
+    if regulation_data[rejected_col].isna().any():
+        raise ValueError(f"Rejected column '{rejected_col}' contains missing values. ")
     mask_rejected = regulation_data[rejected_col].astype(bool)
-    foreground_list = regulation_data.loc[mask_rejected, identifier].unique()
-    background_list = regulation_data.loc[~mask_rejected, identifier].unique()
+    foreground_list = regulation_data.loc[
+        mask_rejected, identifier
+    ]  # .unique()  # ! unique?
```

Comment:
i guess it will be best if we make one of the two the default. Needs to be well explained...